src/main/java/edu/cornell/kfs/module/purap/document/service/impl/CuB2BShoppingServiceImpl.java

```diff
@@ -183,15 +193,47 @@ public List createRequisitionsFromCxml(B2BShoppingCart message, Person user) thr
 
             req.fixItemReferences();
 
-            // save requisition to database
-            purapService.saveDocumentNoValidation(req);
+            if (checkRequisitionAccountsAreUnique(req)) {
+                // save requisition to database
+                purapService.saveDocumentNoValidation(req);
 
-            // add requisition to List
-            requisitions.add(req);
+                // add requisition to List
+                requisitions.add(req);
+            }
+        }
+
+        if (Collections.isEmpty(requisitions)) {
+            throw new B2BShoppingException("Invalid e-shop Request. No requisitions were created.");
         }
+
         return requisitions;
     }
 
+    public boolean checkRequisitionAccountsAreUnique(RequisitionDocument requisitionDocument) {
+        boolean requisitionAccountsUnique = true;
+        uniqueRequisitionAccountValidator = getUniqueRequisitionAccountValidator();
+
+        List<RequisitionItem> requisitionItems = requisitionDocument.getItems();
+        for (RequisitionItem requisitionItem : requisitionItems) {
+            uniqueRequisitionAccountValidator.setItemForValidation(requisitionItem);
+
+            AttributedDocumentEvent attributedDocumentEvent = new AttributedRouteDocumentEvent(requisitionDocument);
+            if (!uniqueRequisitionAccountValidator.validate(attributedDocumentEvent)) {
+                requisitionAccountsUnique = false;
+                handleDuplicateAccountError(requisitionDocument, requisitionItem);
```

Comment:
it looks like if a requisition has multiple items with duplicate account conditions, the user will get multiple email.  I think we want to collect all the error messages, and send just one email per requisition.